lib/gui/modules/image-writer.js

```diff
@@ -117,7 +117,67 @@ imageWriter.service('ImageWriterService', function($q, $timeout, SettingsModel,
    * });
    */
   this.performWrite = function(image, drive, onProgress) {
-    return $q.when(writer.writeImage(image, drive, SettingsModel.data, onProgress));
+    const argv = [ image ];
+    argv.push('--drive', drive.device);
+    argv.push('--robot');
+
+    // Explicitly set the boolen flag in positive
+    // or negative way in order to be on the safe
+    // side in case the Etcher CLI changes the
+    // default value of these options.
+
+    if (SettingsModel.data.unmountOnSuccess) {
+      argv.push('--unmount');
+    } else {
+      argv.push('--no-unmount');
+    }
+
+    if (SettingsModel.data.validateWriteOnSuccess) {
+      argv.push('--check');
+    } else {
+      argv.push('--no-check');
+    }
+
+    return $q(function(resolve, reject) {
+
+      let executable;
+
+      if (isRunningInAsar()) {
+        executable = path.join(process.resourcesPath, 'app.asar');
+      } else {
+        executable = electron.remote.process.argv[1];
+      }
+
+      AnalyticsService.log(`Forking: ${executable} ${argv.join(' ')}`);
+      const child = childProcess.fork(executable, argv, {
+
+        // Pipe stdout/stderr to the parent
+        // We're not using it directly but its
+        // handy for debugging reasons.
+        silent: true
+
+      });
+
+      child.on('message', function(message) {
```

Comment:
Document that we perform IPC communication with the child process. The commit body should be edited to reflect that.